af/shovel/url_intelligence.py

```diff
@@ -0,0 +1,287 @@
+        for row in reader:
+            yield row
+
+def init_category_codes(working_dir, postgres):
+    pgconn = psycopg2.connect(dsn=postgres)
+    with pgconn, pgconn.cursor() as c:
+        csv_path = os.path.join(working_dir, 'test-lists', 'lists', '00-LEGEND-new_category_codes.csv')
+        for row in _iterate_csv(csv_path, skip_header=True):
+            cat_desc, cat_code, cat_old_codes, cat_long_desc = row
+            c.execute('INSERT INTO url_categories (cat_code, cat_desc, cat_long_desc, cat_old_codes)'
+                      ' VALUES (%s, %s, %s, %s)'
+                      ' ON CONFLICT DO NOTHING RETURNING cat_code',
+                      (cat_code, cat_desc, cat_long_desc, cat_old_codes))
+            # XXX maybe we care to know when there is a dup?
+
+special_countries = (
+    ('Unknown Country', 'ZZ', 'ZZZ'),
+    ('Global', 'XX', 'XXX'),
+    ('European Union', 'EU', 'EUE')
+)
+
+def get_country_alpha_2_no(postgres):
+    pgconn = psycopg2.connect(dsn=postgres)
+    with pgconn, pgconn.cursor() as c:
+        c.execute('SELECT alpha_2, country_no FROM countries')
+        country_alpha_2_no = {str(_[0]): _[1] for _ in c}
+    return country_alpha_2_no
+
+def get_cat_code_no(postgres):
+    pgconn = psycopg2.connect(dsn=postgres)
+    with pgconn, pgconn.cursor() as c:
+        c.execute('SELECT cat_code, cat_no FROM url_categories')
+        cat_code_no = {str(_[0]): _[1] for _ in c}
+    return cat_code_no
+
+def init_countries(postgres):
+    pgconn = psycopg2.connect(dsn=postgres)
+    with pgconn, pgconn.cursor() as c:
+        for country in pycountry.countries:
+            alpha_2 = country.alpha_2
+            alpha_3 = country.alpha_3
+            full_name = country.name
+            # This has the nice side-effect of making some names more compact
+            # and "fixing" the "Taiwan, Province of China" issue
+            name = country.name.split(',')[0]
+            c.execute('INSERT INTO countries (name, full_name,  alpha_2, alpha_3)'
+                      ' VALUES (%s, %s, %s)'
+                      ' ON CONFLICT DO NOTHING RETURNING country_no',
+                      (name, full_name, alpha_2, alpha_3))
+
+        for name, alpha_2, alpha_3 in special_countries:
+            c.execute('INSERT INTO countries (name, full_name, alpha_2, alpha_3)'
+                      ' VALUES (%s, %s, %s)'
+                      ' ON CONFLICT DO NOTHING RETURNING country_no',
+                      (name, full_name, alpha_2, alpha_3))
+
+def init_url_lists(working_dir, postgres, cat_code_no, country_alpha_2_no):
+    pgconn = psycopg2.connect(dsn=postgres)
+    with pgconn, pgconn.cursor() as c:
+        csv_glob = os.path.join(working_dir, 'test-lists', 'lists', '*.csv')
+        for csv_path in glob(csv_glob):
+            alpha_2 = os.path.basename(csv_path).split('.csv')[0].upper()
+            if alpha_2 == 'GLOBAL':
+                alpha_2 = 'XX' # We use XX to denote the global country code
```

Comment:
Fun. `ZZ` is "Unknown county", `XX` is "Global". We should probably codify it somewhere.

Reply:
`ZZ` being unknown is documented in OONI spec. `XX` is something internal and specific to the test-lists sync, but I guess we should still document it somewhere.
In any case it's in the ISO code namespace that is reserved for user assigned variables, so it should be safe.